src/module/helpers-dice.js

```diff
@@ -176,44 +202,65 @@ const OseDice = {
     result.target = data.roll.thac0;
     const targetActorData = data.roll.target?.actor?.system || null;
 
-    const targetAc = data.roll.target ? targetActorData.ac.value : 9;
-    const targetAac = data.roll.target ? targetActorData.aac.value : 10;
+    const targetAc = data.roll.target ? targetActorData.ac.value : 20;
+    const targetAac = data.roll.target ? targetActorData.aac.value : -20;
     result.victim = data.roll.target ? data.roll.target.name : null;
 
     if (game.settings.get(game.system.id, "ascendingAC")) {
-      if (
-        (roll.terms[0] != 20 && roll.total < targetAac) ||
-        roll.terms[0] === 1
-      ) {
+      const attackBonus = 19 - data.roll.thac0;
+      if (this.attackIsSuccess(roll, targetAac, attackBonus)) {
+        result.details = game.i18n.format(
+          "OSE.messages.AttackAscendingSuccess",
+          {
+            result: roll.total,
+          }
+        );
+        result.isSuccess = true;
+      } else {
         result.details = game.i18n.format(
           "OSE.messages.AttackAscendingFailure",
           {
             bonus: result.target,
           }
         );
-        return result;
+        result.isFailure = true;
       }
-      result.details = game.i18n.format("OSE.messages.AttackAscendingSuccess", {
-        result: roll.total,
-      });
-      result.isSuccess = true;
-    } else {
-      if (!this.attackIsSuccess(roll, result.target, targetAc)) {
-        result.details = game.i18n.format("OSE.messages.AttackFailure", {
-          bonus: result.target,
-        });
-        return result;
-      }
-      result.isSuccess = true;
+    } else if (this.attackIsSuccess(roll, result.target, targetAc)) {
+      // Answer is bounded betweewn AC -3 and 9 (unarmored) and is shown in chat card
       const value = Math.clamped(result.target - roll.total, -3, 9);
       result.details = game.i18n.format("OSE.messages.AttackSuccess", {
         result: value,
         bonus: result.target,
       });
+      result.isSuccess = true;
+    } else {
+      result.details = game.i18n.format("OSE.messages.AttackFailure", {
+        bonus: result.target,
+      });
+      result.isFailure = true;
     }
     return result;
   },
 
+  // eslint-disable-next-line jsdoc/require-param
```

Comment:
was this included in an earlier commit and not taken out after including the params in a later commit? Or does this JSDoc comment block somehow not fulfill this rule?

Reply:
I couldn't figure it out last time I looked at it, because `param0.speaker` is clearly there. However... There is a missing whitespace between `speaker` and `-`... Fixing that now.